### ISSUE TITLE
Authoring in ES6

### ISSUE THREAD

**Issue by aoberoi** [opentok/opentok-node#131]
Starting this issue to capture any thoughts regarding moving the implementation of this package to ES6.

TL;DR lets wait until node core implements a solution for ECMAScript Modules that is backwards compatible with CommonJS modules. Current draft proposal: https://github.com/nodejs/node-eps/blob/master/002-es6-modules.md

there's a couple concerns around ES6:
1.  module syntax (CommonJS versus ECMAScript Modules)
2.  node version compatibility: we should aim to support at least all node core's supported versions (current, LTS's) but hopefully even further back. node < 6 doesn't support most of ES6 out of the box, the code needs to be in ES5 for anything less than that.

we could address both of these today by adding a compilation step with babel and putting it in the `"prepublish"` script. this would mean that we export a CommonJS module with ES5 syntax,all while authoring in ECMAScript modules with ES6 syntax. this solution isn't all that useful, because the ES6 language features (new data structures like `WeakMap`, usage of `Promise`, and anything a transpiler emits an incomplete or significantly degraded-performance transform for) would not be utilized. Going all in presents us with the ability to release a new and more elegant API, and transpiling ties us to whatever is available in ES5. so lets just wait to go all in.

**Comment by manchuck** [comment]
Closing as we are not doing any enhancements on this SDK. Please consider migrating to the [Vonage Video platform](https://developer.vonage.com/en/video/overview)
